libr/bin/bin.c

```diff
@@ -235,21 +234,19 @@ R_API int r_bin_file_cur_set_plugin(RBinFile *binfile, RBinPlugin *plugin) {
 // maybe too big sometimes? 2KB of stack eaten here..
 #define R_STRING_SCAN_BUFFER_SIZE 2048
 
-static int string_scan_range(RList *list, const ut8 *buf, int min,
+static int string_scan_range(RList *list, RBinFile *bf, int min,
 			      const ut64 from, const ut64 to, int type) {
 	ut8 tmp[R_STRING_SCAN_BUFFER_SIZE];
 	ut64 str_start, needle = from;
 	int count = 0, i, rc, runes;
-	bool rdump = false;
+	const ut8 *buf = bf->buf->buf;
```

Comment:
Its better to use the rbuf api to get the beginning and size of buffer.. because. Abuffer can be something else than a malloc and getters abstract that